lambdas/check-reverification-identity/src/main/java/uk/gov/di/ipv/core/checkreverificationidentity/CheckReverificationIdentityHandler.java

```diff
@@ -0,0 +1,197 @@
+    @Override
+    public Map<String, Object> handleRequest(JourneyRequest request, Context context) {
+        LogHelper.attachComponentId(configService);
+        configService.setFeatureSet(RequestHelper.getFeatureSet(request));
+
+        try {
+            var ipvSessionId = getIpvSessionId(request);
+            LogHelper.attachIpvSessionIdToLogs(ipvSessionId);
+            var ipvSession = ipvSessionService.getIpvSession(ipvSessionId);
+            LogHelper.attachClientSessionIdToLogs(ipvSession.getClientOAuthSessionId());
+            var clientOAuthSession =
+                    clientSessionService.getClientOAuthSession(
+                            ipvSession.getClientOAuthSessionId());
+            LogHelper.attachClientIdToLogs(clientOAuthSession.getClientId());
+            LogHelper.attachGovukSigninJourneyIdToLogs(
+                    clientOAuthSession.getGovukSigninJourneyId());
+
+            var vcs =
+                    evcsService.getVerifiableCredentials(
+                            clientOAuthSession.getUserId(),
+                            clientOAuthSession.getEvcsAccessToken(),
+                            CURRENT);
+
+            if (!vcsContainIdentity(vcs)) {
+                ipvSession.setFailureCode(NO_IDENTITY_AVAILABLE);
+                ipvSessionService.updateIpvSession(ipvSession);
+                return NOT_FOUND_RESPONSE;
+            }
+
+            return FOUND_RESPONSE;
+
+        } catch (HttpResponseExceptionWithErrorBody | EvcsServiceException e) {
+            LOGGER.error(LogHelper.buildErrorMessage(e.getErrorResponse()));
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH, e.getResponseCode(), e.getErrorResponse())
+                    .toObjectMap();
+        } catch (IpvSessionNotFoundException e) {
+            LOGGER.error(LogHelper.buildErrorMessage("IPV session not found", e));
+            return new JourneyErrorResponse(JOURNEY_ERROR_PATH, SC_NOT_FOUND, IPV_SESSION_NOT_FOUND)
+                    .toObjectMap();
+        } catch (CredentialParseException e) {
+            LOGGER.error(LogHelper.buildErrorMessage("Failed to parse credentials", e));
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH,
+                            SC_SERVER_ERROR,
+                            FAILED_TO_PARSE_SUCCESSFUL_VC_STORE_ITEMS)
+                    .toObjectMap();
+        } catch (ParseException e) {
+            LOGGER.error(LogHelper.buildErrorMessage("Failed to get VOT from operational VC", e));
+            return new JourneyErrorResponse(
+                            JOURNEY_ERROR_PATH, SC_SERVER_ERROR, FAILED_TO_PARSE_ISSUED_CREDENTIALS)
+                    .toObjectMap();
+        } catch (Exception e) {
+            LOGGER.error(LogHelper.buildErrorMessage("Unhandled lambda exception", e));
+            throw e;
+        }
+    }
+
+    private boolean vcsContainIdentity(List<VerifiableCredential> vcs)
```

Comment:
This seems very similar to `CheckExistingIdentityHandler#getStrongestAttainedVotForVtr` (with all valid VOTs and no CIs) - do you think there's any merit in trying to commonise the two? Either by merging the lambdas or pulling out a shared service...

There might be a question about whether we _need_ to check for a profile here at all - in principle we should only have an identity in EVCS if they match(ed) a valid profile. It probably doesn't hurt - and should be fairly speedy (if we don't call CIMIT), so I'm not wildly fussed.

Reply:
They seem similar because I used it to inform this code! I figured that they were different enough though to stand on their own. Can look at pulling out a common service but I suspect it may be more trouble than it's worth.

I'd rather not combine the two lambdas - check-existing-identity is already pretty complicated and sprawling, and shoehorning another code path through it gives me the fear.